soepy/python/pre_processing/model_processing.py

```diff
@@ -111,8 +127,18 @@ def group_parameters(init_dict, init_dict_flat):
         init_dict["PARAMETERS"]["delta_s3"],
     )
 
-    init_dict_flat["theta_p"] = init_dict["PARAMETERS"]["theta_p"]
-    init_dict_flat["theta_f"] = init_dict["PARAMETERS"]["theta_f"]
+    if init_dict["DERIVED_ATTR"]["num_types"] > 1:
+        for i in ["p", "f"]:
+            init_dict_flat["theta_" + i] = [
+                v for k, v in init_dict["PARAMETERS"].items() if str("theta_" + i) in k
```

Comment:
Please remove use of `str()` everywhere.

Reply:
Hopefully done.